langchain/vectorstores/weaviate.py

```diff
@@ -137,19 +139,18 @@ def json_serializable(value: Any) -> Any:
                         data_properties[key] = json_serializable(metadatas[i][key])
 
                 # If the UUID of one of the objects already exists
-                # then the existing objectwill be replaced by the new object.
+                # then the existing object will be replaced by the new object.
                 if "uuids" in kwargs:
                     _id = kwargs["uuids"][i]
                 else:
                     _id = get_valid_uuid(uuid4())
 
                 if self._embedding is not None:
-                    embeddings = self._embedding.embed_documents(list(doc))
```

Comment:
great catch! i wonder if we should just change list(doc) -> [doc] and leave the rest the same. reason being current implementation lets us lazily load in texts, whereas calling list(texts) up front would load all of them into memory

Reply:
> great catch! i wonder if we should just change list(doc) -> [doc] and leave the rest the same. reason being current implementation lets us lazily load in texts, whereas calling list(texts) up front would load all of them into memory

I checked the `add_texts` methods of all vectore stores. Here is what I found:

1. **Twelve** vector stores turn the potential iterable `texts` into a list first and then embed them:
* chroma
* pgvector
* qdrant
* supabase
* analyticdb
* atlas
* deeplake
* elastic_vector_search
* lancedb
* milvus
* opensearch_vector_search
* tair

2. **Four** vector stores that iterate over `texts` and embed text one by one lazily:
* faiss
* redis
* pinecone
* myscale

The behavious are not consistent. But most vector stores simply turn the `texts` variable into a list first.

Reply:
yea we definitely haven't been consistent about it, which is on us. and it's possible for a lot of workloads / embedding methods that embedding all the texts at once is more efficient. but think i'd prefer to not change behavior in an existing vector store until we've come up with a best practice that we apply everywhere. meaning in this case slight preference to keep lazy for now (otherwise somebody who's using Weaviate today could see their memory usage change next update for seemingly no reason)